### PR TITLE
Add support for random MAC address

### PR DESCRIPTION
**Pull request by krish2718**
Add a build time option to generate random MAC address, this helps in test deployments.

Used Zephyr API in the BM code to simplify things, as srand() was giving same MAC address (seed problems).

### CODE REVIEW

File: nrf70_bm_lib/docs/source/nrf70_bm_porting_guide.rst [hunk 1]
```diff
@@ -47,6 +47,10 @@ The reference implementation of the BM Driver for the Zephyr RTOS uses build-tim
     - Used to manage the timers for the nRF70 Series driver, esp. for low power mode.
     - timer_***
     - k_work_delayable_***
+  * - PRNG
```
Comment (ioannisg) [inline nrf70_bm_lib/docs/source/nrf70_bm_porting_guide.rst:50]: Would be nice to write the full name for PRNG - not just the acronym

File: nrf70_bm_lib/docs/source/nrf70_bm_porting_guide.rst [hunk 2]
```diff
@@ -47,6 +47,10 @@ The reference implementation of the BM Driver for the Zephyr RTOS uses build-tim
...
+    - Used to generate random numbers for the nRF70 Series random MAC address generation.
```
Comment (ioannisg) [inline nrf70_bm_lib/docs/source/nrf70_bm_porting_guide.rst:51]:
```suggestion
    - Used to generate random numbers for the nRF70 Series random MAC address generation (if random MAC address generation support is enabled).
```

**Review by ioannisg — APPROVED** [review]
Looks good to me. Maybe in the documentation you can write a very short paragraph about the options for the MAC address setting - that would help the developers a lot, IMHO. They would not need to look into the Kconfig setting.

**Comment by krish2718** [comment]
> Looks good to me. Maybe in the documentation you can write a very short paragraph about the options for the MAC address setting - that would help the developers a lot, IMHO. They would not need to look into the Kconfig setting.

Done, added a separate section.

**Review by ioannisg — APPROVED** [review]
Thanks! @sachinthegreen pls, review the code part as well :)

**Pull request merged**